### PR TITLE
Added offline image download capability using mergebin

### PR DESCRIPTION
**Pull request by makermelissa**

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -233,10 +233,18 @@
...
-  #app.connected #commands {
+  #app.connected #commands, #app.gererating #commands {
     height: 150px;
   }
 
+  #app.connected li.offline {
+    display: none;
+  }
+
+  #app.gererating li.online {
```
Comment (lorennorman) [inline css/style.css]: Is `gererating` a typo for `generating`? (it's matched in the `.js` file as well if you decide to change it)

File: js/script.js [hunk 2]
```diff
@@ -112,8 +120,28 @@ document.addEventListener("DOMContentLoaded", () => {
...
+                `  - Look for a little black button near the power port.\n` +
+                `2. Refresh your browser and try again.\n` +
+                `3. Make sure you are not connected in another browser tab.\n` +
+                `4. Double-check your board type and serial port selection.\n` +
+                `5. Post on the Support Forum (link above) with this info:\n\n` +
+                `"Firmware Tool: ${e}"\n`
+            );
+            await disconnect();
+            toggleUIConnected(false);
+        });*/
```
Comment (lorennorman) [inline js/script.js]: what's this commented code hanging around for?

File: js/script.js [hunk 3]
```diff
@@ -737,58 +779,70 @@ async function mergeSettings() {
...
+        return;
     }
 
     // pretty print the settings object with VERSION placeholders filled
-    const settings = await mergeSettings()
-    const settingsString = JSON.stringify(settings, null, 2)
-    const strippedSettings = settingsString.replaceAll('VERSION', semver)
-    logMsg(`Flashing with settings: <pre>${strippedSettings}</pre>`)
+    const settings = await mergeSettings();
+    console.log(settings);
```
Comment (lorennorman) [inline js/script.js]: Not sure if you're leaving console logs in intentionally? If not, there's usage further down.

**Review by lorennorman — APPROVED** [review]
💯 Looks good! I'm impressed at how little HTML you needed. My comments are simple things that may not even need changing.

Aside, I wonder if there's an easy way to batch up stylistic changes (whitespace, semicolons, etc) together and somehow exclude them from the main PR diff under review?

File: css/style.css (hunk 1 above)
Comment (makermelissa) [reply css/style.css]: Yep, that's a typo. Good catch.

File: js/script.js (hunk 2 above)
Comment (makermelissa) [reply js/script.js]: I was debugging. I need to uncomment.

File: js/script.js (hunk 3 above)
Comment (makermelissa) [reply js/script.js]: Yeah, I meant to remove.

**Comment by makermelissa** [comment]
> 💯 Looks good! I'm impressed at how little HTML you needed. My comments are simple things that may not even need changing.
 
Thanks.

> Aside, I wonder if there's an easy way to batch up stylistic changes (whitespace, semicolons, etc) together and somehow exclude them from the main PR diff under review?

I'm not sure if that's possible. Honestly though, the style updates were really needed and I probably should have done those a while back.

**Comment by lorennorman** [comment]
>  the style updates were really needed

How so? IDE screaming at you or something?

**Comment by makermelissa** [comment]
> How so? IDE screaming at you or something?

No, it didn't need to scream. It just did them automatically. ;) 

However, it made the code much more readable. Also, I blame myself for the spacing being off in most places because I originally wrote it on glitch, where the editor has a default tab spacing of 2 characters and I hadn't run it through a formatter when moving over to github.

**Pull request merged**